developer/src/tike/xml/layoutbuilder/constants.js

```diff
@@ -672,14 +672,16 @@ $(function() {
 
   this.showAllModifierCombinations = false;
 
-  // Add CAPS and NO_CAPS variants for all of the above
-  (function(c, codes) {
-    var i, n = c.length;
-    for(i = 0; i < n; i++)
-      c.push(c[i] | codes.CAPS);
-    for(i = 0; i < n; i++)
-      c.push(c[i] | codes.NO_CAPS);
-  })(this.validModifierCombinations, this.modifierCodes);
+  // Add CAPS variants for all of the above
+  (function addCapsCombinations(validCombinations, CAPS) {
+    const n = validCombinations.length;
+    for(let i = 0; i < n; i++) {
+      const newCombination = validCombinations[i] | CAPS;
+      if(!validCombinations.includes(newCombination)) {
+        validCombinations.push(newCombination);
+      }
```

Comment:
I tweaked this here because, for example, `caps` was in the list three times previously.

Reply:
Is `nocaps` at l.696 still used?

Reply:
It's not, but I left it there for completeness because it maps to the definitions in kmn